### PR TITLE
RavenDB-5090 Create export database endpoint

### PR DESCRIPTION
**Pull request by IdanHaim**
* include RavenDB-5089 Create validateExportOptions endpoint

### CODE REVIEW

File: src/Raven.Server/Smuggler/Documents/SmugglerExporter.cs [hunk 1]
```diff
@@ -50,20 +50,47 @@ public ExportResult Export(DocumentsOperationContext context, Stream destination
...
+                    onProgress?.Invoke(progress);
                     writer.WriteComma();
                     writer.WritePropertyName(("Docs"));
-                    var documents = DocumentsLimit.HasValue
-                        ? _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0, 0, DocumentsLimit.Value)
-                        : _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0);
+
+                    IEnumerable<Document> documents = Options.CollectionsToExport.Count != 0 ? 
+                        _database.DocumentsStorage.GetDocumentsFrom(context, Options.CollectionsToExport, StartDocsEtag ?? 0, Options.BatchSize) : 
+                        _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0, 0, Options.BatchSize);
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/SmugglerExporter.cs:62]: Do we still need BatchSize? What will happen if we will have more then BatchSize (1024) docs?

**Comment by ayende** [comment]
Tests are failing because of this.

**Comment by ppekrol** [comment]
@ml054 Please review notification/progress and Studio part

**Comment by ml054** [comment]
Studio review completed.

**Pull request merged**